core/store/src/cold_storage.rs

```diff
@@ -24,6 +24,7 @@ struct StoreWithCache<'a> {
 
 /// The BatchTransaction can be used to write multiple set operations to the cold db in batches.
 /// [`write`] is called every time `transaction_size` overgrows `threshold_transaction_size`.
+/// [`write`] should also be called manually before dropping BatchTransaction to write any leftovers.
```

Comment:
Why not the drop trait? Then it would happen automatically.

Reply:
It is already idiomatic that we manually write transactions before they are dropped.
I think the way to minimise confusion is to go other direction and make all writes "manual". Like, add write_if_full method that calls for write if batch is big enough and call it manually after set.
Not necessarily that clumsy, but it may be nice to just identify in the name of set that set may be calling write, so it is clear to the developer who didn't first read the BatchTransaction documentation.